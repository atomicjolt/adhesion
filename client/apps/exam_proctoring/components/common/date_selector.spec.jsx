import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import DateSelector from './date_selector';

describe('Date Selector', () => {
  let props;
  let result;
  const date = moment('2010-10-20').toDate();

  beforeEach(() => {
    props = {
      date,
      onChange: () => {},
    };
    result = shallow(<DateSelector {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
