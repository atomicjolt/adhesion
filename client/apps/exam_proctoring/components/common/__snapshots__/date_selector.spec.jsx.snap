// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Selector matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={undefined}
>
    <div
        style={
            Object {
                "position": "relative",
              }
        }
    >
        <DatePicker
            dateFormat="ddd MMM Do YYYY"
            dateFormatCalendar="MMMM YYYY"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            popoverAttachment="top left"
            popoverTargetAttachment="bottom left"
            popoverTargetOffset="10px 0"
            readOnly={true}
            selected={"2010-10-20T06:00:00.000Z"}
            tetherConstraints={
                Array [
                    Object {
                      "attachment": "together",
                      "to": "window",
                    },
                  ]
            }
        />
        <button
            className="qa-calendar-btn"
            onMouseDown={[Function]}
            style={
                Object {
                    "backgroundColor": "white",
                    "border": "none",
                    "color": "#555555",
                    "padding": "0px",
                    "position": "absolute",
                    "right": "5px",
                    "top": "7px",
                  }
            }
        >
            <i
                className="material-icons"
            >
                today
            </i>
        </button>
    </div>
</div>,
  "nodes": Array [
    <div
      style={undefined}
>
      <div
            style={
                  Object {
                        "position": "relative",
                      }
            }
      >
            <DatePicker
                  dateFormat="ddd MMM Do YYYY"
                  dateFormatCalendar="MMMM YYYY"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  popoverAttachment="top left"
                  popoverTargetAttachment="bottom left"
                  popoverTargetOffset="10px 0"
                  readOnly={true}
                  selected={"2010-10-20T06:00:00.000Z"}
                  tetherConstraints={
                        Array [
                              Object {
                                "attachment": "together",
                                "to": "window",
                              },
                            ]
                  }
            />
            <button
                  className="qa-calendar-btn"
                  onMouseDown={[Function]}
                  style={
                        Object {
                              "backgroundColor": "white",
                              "border": "none",
                              "color": "#555555",
                              "padding": "0px",
                              "position": "absolute",
                              "right": "5px",
                              "top": "7px",
                            }
                  }
            >
                  <i
                        className="material-icons"
                  >
                        today
                  </i>
            </button>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DateSelector
        date={2010-10-20T06:00:00.000Z}
        onChange={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DateSelector {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "date": 2010-10-20T06:00:00.000Z,
          "onChange": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={undefined}
>
          <div
                    style={
                              Object {
                                        "position": "relative",
                                      }
                    }
          >
                    <DatePicker
                              dateFormat="ddd MMM Do YYYY"
                              dateFormatCalendar="MMMM YYYY"
                              disabled={false}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              popoverAttachment="top left"
                              popoverTargetAttachment="bottom left"
                              popoverTargetOffset="10px 0"
                              readOnly={true}
                              selected={"2010-10-20T06:00:00.000Z"}
                              tetherConstraints={
                                        Array [
                                                  Object {
                                                    "attachment": "together",
                                                    "to": "window",
                                                  },
                                                ]
                              }
                    />
                    <button
                              className="qa-calendar-btn"
                              onMouseDown={[Function]}
                              style={
                                        Object {
                                                  "backgroundColor": "white",
                                                  "border": "none",
                                                  "color": "#555555",
                                                  "padding": "0px",
                                                  "position": "absolute",
                                                  "right": "5px",
                                                  "top": "7px",
                                                }
                              }
                    >
                              <i
                                        className="material-icons"
                              >
                                        today
                              </i>
                    </button>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={undefined}
>
          <div
                    style={
                              Object {
                                        "position": "relative",
                                      }
                    }
          >
                    <DatePicker
                              dateFormat="ddd MMM Do YYYY"
                              dateFormatCalendar="MMMM YYYY"
                              disabled={false}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              popoverAttachment="top left"
                              popoverTargetAttachment="bottom left"
                              popoverTargetOffset="10px 0"
                              readOnly={true}
                              selected={"2010-10-20T06:00:00.000Z"}
                              tetherConstraints={
                                        Array [
                                                  Object {
                                                    "attachment": "together",
                                                    "to": "window",
                                                  },
                                                ]
                              }
                    />
                    <button
                              className="qa-calendar-btn"
                              onMouseDown={[Function]}
                              style={
                                        Object {
                                                  "backgroundColor": "white",
                                                  "border": "none",
                                                  "color": "#555555",
                                                  "padding": "0px",
                                                  "position": "absolute",
                                                  "right": "5px",
                                                  "top": "7px",
                                                }
                              }
                    >
                              <i
                                        className="material-icons"
                              >
                                        today
                              </i>
                    </button>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DateSelector
    date={2010-10-20T06:00:00.000Z}
    onChange={[Function]}
/>,
}
`;
